### ISSUE TITLE
FAQ links dependent on route to details page

### ISSUE THREAD

**Issue by MikeMcC399** [corona-warn-app/cwa-website#2748]
## Where to find the issue

- https://www.coronawarn.app/en/faq/results/
- https://www.coronawarn.app/de/faq/results/

## Describe the issue

FAQ topic links depend on which route was taken to the FAQ details screen https://www.coronawarn.app/en/faq/results/

If the "View all topics and all questions" directly under the Search button is used to access https://www.coronawarn.app/en/faq/results then topic links, such as https://www.coronawarn.app/en/faq/results/#app_features_risk_assessment are direct links to an anchor.

If on the other hand "View all topics and all questions" at the bottom of a column of lists of topics is used to access https://www.coronawarn.app/en/faq/results then topic links, such as https://www.coronawarn.app/en/faq/results/?search=&topic=application#app_features_risk_assessment use a search format, not a direct link to an anchor.

## Steps to reproduce

1. Open https://www.coronawarn.app/en/faq/ in Google Chrome desktop
2. Click on "View all topics and all questions" directly under Search button (links to https://www.coronawarn.app/en/faq/results)<br>![Step 2](https://user-images.githubusercontent.com/66998419/164403809-045810d6-23a4-4d98-9d61-1d2c89cc4cb2.jpg)
3. Hover over "Risk Assessment" in left-hand menu<br>
![Step 3](https://user-images.githubusercontent.com/66998419/164403900-24ca4e86-fe51-451f-a899-28de98d330ad.jpg)
4. Note link is https://www.coronawarn.app/en/faq/results/#app_features_risk_assessment
5. Return to https://www.coronawarn.app/en/faq/
6. Click on "View all topics and all questions" at the bottom of the "App Features" column (links to https://www.coronawarn.app/en/faq/results/?search=&topic=application)<br>![Step 6](https://user-images.githubusercontent.com/66998419/164403959-1fd1a0b1-09b0-48c7-8845-7a7d90ac59b1.jpg)
7. Hover over "Risk Assessment" in left-hand menu
8. Note link is https://www.coronawarn.app/en/faq/results/?search=&topic=application#app_features_risk_assessment<br>
![Step 8](https://user-images.githubusercontent.com/66998419/164404055-4709b94a-ba07-4f68-97eb-b495303c97b7.jpg)

## Expected behaviour

Make the behaviour of the page https://www.coronawarn.app/en/faq/results/ independent of the method used to reach it.

Preferably, topics on the left-hand menu should use direct links such as https://www.coronawarn.app/en/faq/results/#app_features_risk_assessment instead of search links such as https://www.coronawarn.app/en/faq/results/?search=&topic=application#app_features_risk_assessment

**Comment by MikeMcC399** [comment]
@svengabr 

Unfortunately that solution does not cover the case where https://www.coronawarn.app/en/faq/results/ is opened directly e.g. from a bookmark, so that would not be a preferred solution in my eyes.

It would also make it more difficult to bookmark a clean hyperlink (without a search term in it) such as https://www.coronawarn.app/en/faq/results/#app_features_risk_assessment.

**Comment by brifemu** [comment]
Morning @MikeMcC399, in one of the last PR this looks to be fixed and now this is not reproducible at [coronawarn.app](https://coronawarn.app). Can you confirm it? Thanks!

**Comment by MikeMcC399** [comment]
Good morning @brifemu !

> in one of the last PR this looks to be fixed and now this is not reproducible at [coronawarn.app](https://coronawarn.app/). Can you confirm it?

I checked and the issue I described is still reproducible. To make it clearer, I added screenshots to the steps to reproduce. Please let me know if anything is still unclear.

The usability of the FAQ has improved a lot since the migration to the new layout. I hope that we can also remove the remaining issues.

**Comment by dsarkar** [comment]
Regarding point 7 and 8: Now it links to URL https://www.coronawarn.app/en/faq/results/#app_features_risk_assessment

**Comment by MikeMcC399** [comment]
@dsarkar 

> Regarding point 7 and 8: Now it links to URL https://www.coronawarn.app/en/faq/results/#app_features_risk_assessment as in 3 and 4.

That is quite strange, because I don't see that. I tried in Google Chrome and Mozilla Firefox under Windows 11. In both browsers I get https://www.coronawarn.app/en/faq/results/?search=&topic=application#app_features_risk_assessment, not the link https://www.coronawarn.app/en/faq/results/#app_features_risk_assessment which you posted.

**Comment by dsarkar** [comment]
<img width="401" alt="image" src="https://user-images.githubusercontent.com/1409741/164514441-3aba6741-d03d-4e62-9a2d-3487b91969d9.png">

**Comment by MikeMcC399** [comment]
@dsarkar 

Your second screen shot looks like it is from https://www.coronawarn.app/en/faq/. That is not the page with the problem.

Please scroll down to the bottom of that page and click on 
"View all topics and all
questions"
![image](https://user-images.githubusercontent.com/66998419/164515291-9ca3411d-55ec-43df-bfa7-15c7d72e94e2.png)

You should be taken to URL https://www.coronawarn.app/en/faq/results/?search=&topic=application

Now click on Risk Assessment on the left and the URL changes to https://www.coronawarn.app/en/faq/results/?search=&topic=application#app_features_risk_assessment

If that is still unclear, I can give you a Cypress test to automate the steps. (I decided it would be helpful to learn a bit of Cypress 🙂!)

**Comment by dsarkar** [comment]
@MikeMcC399 Ok, I see. You are right.

**Comment by MikeMcC399** [comment]
@dsarkar

Thank you very much for following through on this issue! I have reviewed it and I think we should close it as "working as designed".

There were several other issues interacting with one another which made the handling of links confusing:
- https://github.com/corona-warn-app/cwa-website/issues/2735
- https://github.com/corona-warn-app/cwa-website/issues/2771
- https://github.com/corona-warn-app/cwa-website/issues/2772
- https://github.com/corona-warn-app/cwa-website/issues/2779

Now that the above and other issues have been resolved, the FAQ seems to be working correctly.

The different URLs result from filtering. This is not in itself incorrect. If there are any other undesired side-effects resulting from this, we should handle them in separate issues.
